GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs

```diff
@@ -53,50 +72,58 @@ public bool Load(List<string> scriptLocations)
             var assemblyName = Path.GetRandomFileName();
 
             var references = new List<MetadataReference>();
-            foreach (var a in AppDomain.CurrentDomain.GetAssemblies())
-                if (!a.IsDynamic && !a.Location.Equals(""))
-                    references.Add(MetadataReference.CreateFromFile(a.Location));
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                if (!assembly.IsDynamic && !assembly.Location.Equals(""))
+                {
+                    references.Add(MetadataReference.CreateFromFile(assembly.Location));
+                }
+            }
+            //Add vector2 reference because vector2 isn't in System.Numerics in dot net core
+            references.Add(MetadataReference.CreateFromFile(typeof(Vector2).Assembly.Location));
             //Now add game reference
             references.Add(MetadataReference.CreateFromFile(typeof(Game).Assembly.Location));
-            var op = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
+            var compilationOptions = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
                 .WithOptimizationLevel(OptimizationLevel.Release).WithConcurrentBuild(true);
```

Comment:
Unsure if this has been a hindrance, but it can be addressed in this PR so may as well.

Reply:
What are you referring to as a hindrance?

Reply:
If you mean the name change, acronyms and abbreviations should be avoided wherever possible. I like to clean up files I work in.